src/ga/population.rs

```diff
@@ -139,6 +141,46 @@ impl PopulationGenerator<Vec<bool>> for BitStrings {
   }
 }
 
+/// Implements [PopulationGenerator] trait. Can be used with genetic algorithm.
+///
+/// Generates random permutations of given vector.
+/// Permutations can be repeated
+pub struct RandomPermutations<GeneT: Copy> {
+  genes: Vec<GeneT>,
+}
+
+impl<GeneT: Copy> RandomPermutations<GeneT> {
+  /// Returns [RandomPermutations] population generator.
+  ///
+  /// ### Arguments
+  ///
+  /// * `genes` - Vector which will be permuted
+  pub fn new(genes: Vec<GeneT>) -> Self {
+    RandomPermutations { genes }
+  }
+}
+
+impl<GeneT: Copy + Debug + Sync + Send> PopulationGenerator<Vec<GeneT>> for RandomPermutations<GeneT> {
```

Comment:
```suggestion
impl<GeneT: Copy> PopulationGenerator<Vec<GeneT>> for RandomPermutations<GeneT> {
```

Is there a particular reason for these three trait bounds: `Debug`, `Sync`, `Send`?

Particularly the last two should be auto-implemented (you don't need to do it manually AFAIK.

Reply:
It didn't work without them.